File: react-app/src/components/LikeButton.js
import React, { useState } from "react";

function LikeButton() {
    const [count, setCount] = useState(0);

    return (
        <div>
            <p>Likes: {count}</p>
            <button onClick={() => setCount(count + 1)}>Like</button>
        </div>
    );
}

export default LikeButton;

File: rust-yew/src/components/like_button.rs
use yew::prelude::*;

#[function_component(LikeButton)]
pub fn like_button() -> Html {
    let count = use_state(|| 0);

    let onclick = {
        let count = count.clone();
        Callback::from(move |_| count.set(*count + 1))
    };

    html! {
        <div>
            <p>{ format!("Likes: {}", *count) }</p>
            <button {onclick}>{ "Like" }</button>
        </div>
    }
}
